### PR TITLE
Initial github CI

### PR DESCRIPTION
**Pull request by micprog**
Move open-source CI components to github actions

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +54,7 @@ XVLOG_ARGS += -64bit -compile -vtimescale 1ns/1ns -quiet
 
 define generate_vsim
 	echo 'set ROOT [file normalize [file dirname [info script]]/$3]' > $1
-	bender script vsim --vlog-arg="$(VLOG_ARGS)" $2 | grep -v "set ROOT" >> $1
+	$(BENDER) script vsim --vlog-arg="$(VLOG_ARGS)" $2 | grep -v "set ROOT" >> $1
```
Comment (thommythomaso) [inline Makefile:57]: Is `bender` here as a default correct? or should the default be `./bender`?

File: Makefile [hunk 2]
```diff
@@ -258,6 +258,17 @@ gen_ci: .gitlab-ci.yml
 ci_clean:
 	rm -f .gitlab-ci.yml
 
+bender:
```
Comment (thommythomaso) [inline Makefile:261]: Is this part of the `CI`-section of the makefile? or should we put this in a dedicated section?

**Review by thommythomaso — COMMENTED** [review]
Awesome work! Thanks @micprog for allowing the checks and the doc to run in the GitHub CI.

File: Makefile (hunk 2 above)
Comment (micprog) [reply Makefile:261]: Currently this is only used in the CI, but I'm open to moving it to the main section and updating the `BENDER` variable on top. Updating the top variable would probably also be improved if the necessary make targets have the bender target as a prerequisite.

**Review by thommythomaso — APPROVED** [review]

**Pull request merged**